### PR TITLE
Add search field

### PR DESCRIPTION
**Pull request by steverydz**
## Done
Added snaps search

## QA
- Go to https://snapcraft-io-3534.demos.haus/admin/ahnuP3quahti9vis8aiw/snaps
- Click the "Add snap" button in the top right of the app
- Search for a snap (the list should appear after you have typed two characters)
- Click on any of the results and check that it is added to a list beneath the search box
- Click the bin icon in the top right of any snap in that list and check that it is removed
- Click the "Add snaps" button at the bottom of the column and check that those snaps are added to the snaps table
- Click the bin icon at the end of the table row for any snaps you have added and check that it is removed

## Issue
Fixes https://github.com/canonical-web-and-design/snap-squad/issues/2029

### CODE REVIEW

**Comment by steverydz** [comment]
@steverydz:

> the change to search in all stores for snaps to include is now in production (query would be something like: `/api/v2/stores/the-store-id/snaps?q=example&allowed-for-inclusion=1)`

**Review by sowasred2012 — CHANGES_REQUESTED** [review]
This is on Chrome & Firefox: if I search for a snap, click it and bin it, then search for another snap and click it, I see the snap I previously binned:
search

**Comment by steverydz** [comment]
@sowasred2012 I've fixed the issue you found.

File: webapp/admin/views.py [hunk 1]
```diff
@@ -42,6 +43,24 @@ def get_stores():
     )
 
 
+@admin.route("/admin/<store_id>/snaps/search")
```
Comment (tbille) [inline webapp/admin/views.py:46]:
```suggestion
@admin.route("/admin/<store_id>/snaps/search.json")
```

I like when we add the .json at then, it helps understanding what am I going to get as a return value

File: static/js/public/manage-snaps.js [hunk 2]
```diff
@@ -22,32 +25,144 @@ function handleSnapsFilter(STATE) {
...
 
+function handleSnapsSearch(STATE) {
+  const searchField = document.querySelector("#autoComplete");
+
+  // eslint-disable-next-line
+  const autoCompleteJS = new autoComplete({
+    data: {
+      src: async () => {
+        const source = await fetch(
+          `/admin/${STATE.store.id}/snaps/search?q=${searchField.value}&allowed_for_inclusion=${STATE.store.id}`
```
Comment (tbille) [inline static/js/public/manage-snaps.js:47]:
```suggestion
          `/admin/${STATE.store.id}/snaps/search.json?q=${searchField.value}&allowed_for_inclusion=${STATE.store.id}`
```

**Review by tbille — APPROVED** [review]
LGTM

**Comment by steverydz** [comment]
From @danielmutis (cc @AnnSoMuller)

Screenshot 2021-05-21 at 09 23 43

Screenshot 2021-05-21 at 09 23 53

**Comment by AnnSoMuller** [comment]
@steverydz 
Seems ok for me.

Just when I add a snap that is already there I have an error that makes me confused:
Screenshot 2021-05-21 at 10 27 37
Were other snaps well added? What was the pb etc.
Can we remove error message when uploading a snaps that is already in the list? 
It is ok to have a "success" message as the result is that the snap is in the list (of course just once)

**Comment by steverydz** [comment]
Issue made to address the above comment https://github.com/canonical-web-and-design/snapcraft.io/issues/3551

**Pull request merged**